### DISCUSSION TITLE
add dependencies  --exclude

### DISCUSSION THREAD

**Discussion by heyfavour** [fastapi/fastapi#6980] [Questions]
i  add a golbal dependencies in my project,but there is one api don't need this dependencies .so how can dependencies exlude that api pythonicly.
i read the add_api_route code, it always  inherits the parent router dependencies。
i think you can add a parameter to make parent router dependencies=[]
or can i do something to  make dependencies  exlude that api

**Comment by heyfavour** [comment]
i think it better add parameter in sub api instead of parent api,when parent api extend sub api,check the parent dependencies and set dependencies=[ ]

**Comment by tiangolo** [comment]
Sorry, I can't help you if you don't follow the issue template and provide more information.

If you still have problems please create a new issue and follow the template step by step.

Given that, I'm gonna close this one.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
